File: soyeon/[3] Programmers/가장 큰 수.java
import java.util.*;
class Solution {
    public String solution(int[] numbers) {
    	String str[]=new String[numbers.length];
    	for(int i=0;i<numbers.length;i++) {
    		str[i]=String.valueOf(numbers[i]);
    	}
    	//두개를 이어붙일 때 더 큰쪽으로 정렬되도록
    	Arrays.sort(str, new Comparator<String>() {
            @Override
            public int compare(String s1, String s2) {
                return (s2+s1).compareTo(s1+s2);
            }
        });
    	
    	//이부분 주의, 0으로만 구성되어있으면 0 return
    	if (str[0].equals("0")) return "0";
    	
        String answer = "";
        for(int i=0;i<numbers.length;i++) {
        	answer+=str[i];
        }
        return answer;
    }
}

File: kuyhochung/5. 프로그래머스/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(vector<int> numbers) {
    string answer = "";
    
    vector <pair <int, int> > compare;
    
    for (int i = 0; i < numbers.size(); i++) {
        int original = numbers[i];
        while (numbers[i] < 1000 && numbers[i] != 0) {
            if (numbers[i] < 10) {
                numbers[i] = numbers[i] * 10 + numbers[i];
            }
            else if (numbers[i] < 100) {
                numbers[i] = numbers[i] * 100 + numbers[i];
            }
            else {
                numbers[i] = numbers[i]*10 + numbers[i]/100;
            }
        }
        compare.push_back(make_pair(numbers[i], original));
    }
    sort(compare.begin(),compare.end());
    
    while(!compare.empty()) {
        if (answer == "0" && compare.back().second == 0) {
            compare.pop_back();
            continue;
        }
        answer += to_string(compare.back().second);
        compare.pop_back();
    }
    
    return answer;
}